omx/gstomxaudioenc.c

```diff
@@ -822,21 +865,48 @@ gst_omx_audio_enc_set_format (GstAudioEncoder * encoder, GstAudioInfo * info)
       return FALSE;
   } else {
     if (!(klass->cdata.hacks & GST_OMX_HACK_NO_DISABLE_OUTPORT)) {
-      /* Disable output port */
-      if (gst_omx_port_set_enabled (self->enc_out_port, FALSE) != OMX_ErrorNone)
+      GST_DEBUG_OBJECT (self, "Enabling ports");
+
+      /* Enable input port */
+      if (gst_omx_port_set_enabled (self->enc_in_port, TRUE) != OMX_ErrorNone)
+        return FALSE;
+
+      if (gst_omx_port_wait_enabled (self->enc_in_port,
+              1 * GST_SECOND) != OMX_ErrorNone)
+        return FALSE;
+
+      /* Enable output port */
+      if (gst_omx_port_set_enabled (self->enc_out_port, TRUE) != OMX_ErrorNone)
         return FALSE;
 
       if (gst_omx_port_wait_enabled (self->enc_out_port,
               1 * GST_SECOND) != OMX_ErrorNone)
         return FALSE;
 
+      gst_omx_port_get_port_definition (self->enc_in_port, &port_def_in);
+      GST_DEBUG_OBJECT (self,
+          "Input port definition: nBufferCountActual=%d, format.audio.eEncoding=%d (OMX_AUDIO_CodingPCM=%d), nPortIndex=%d, nBufferSize=%d",
+          (int) port_def_in.nBufferCountActual,
+          (int) port_def_in.format.audio.eEncoding, (int) OMX_AUDIO_CodingPCM,
+          (int) port_def_in.nPortIndex, (int) port_def_in.nBufferSize);
+      gst_omx_port_get_port_definition (self->enc_out_port, &port_def_out);
+      GST_DEBUG_OBJECT (self,
+          "Output port definition: nBufferCountActual=%d, format.audio.eEncoding=%d (OMX_AUDIO_CodingAAC=%d), nPortIndex=%d, nBufferSize=%d",
+          (int) port_def_out.nBufferCountActual,
+          (int) port_def_out.format.audio.eEncoding, (int) OMX_AUDIO_CodingAAC,
+          (int) port_def_out.nPortIndex, (int) port_def_out.nBufferSize);
+
+      GST_DEBUG_OBJECT (self, "Setting Idle state");
       if (gst_omx_component_set_state (self->enc,
               OMX_StateIdle) != OMX_ErrorNone)
         return FALSE;
 
+      GST_DEBUG_OBJECT (self, "Allocating buffers");
       /* Need to allocate buffers to reach Idle state */
       if (gst_omx_port_allocate_buffers (self->enc_in_port) != OMX_ErrorNone)
         return FALSE;
+      if (gst_omx_port_allocate_buffers (self->enc_out_port) != OMX_ErrorNone)
```

Comment:
How was this working previously??

Reply:
This class was made for other platforms and when I tried to run it on DM8168 EVM it just didn't work, so in order to set the component properly I also had to allocate buffers for the output port.